python/plugins/processing/gui/wrappers.py

```diff
@@ -936,7 +936,13 @@ def createWidget(self):
             self.context = dataobjects.createContext()
 
             try:
-                self.combo.setLayer(iface.activeLayer())
+                if self.parameterDefinition().flags() & QgsProcessingParameterDefinition.FlagOptional:
+                    self.combo.setLayer(self.parameterDefinition().defaultValue())
```

Comment:
This isn't going to work everywhere -- the default value may be a string pointing to a filename for instance, whereas setLayer requires a map layer value.

Reply:
Replaced with `setValue` which should handle all of the value types